src/sources/kafka.rs

```diff
@@ -102,16 +103,15 @@ fn kafka_source(
         let stream = OwnedConsumerStream {
             upstream: OwningHandle::new_with_fn(consumer, |c| {
                 let cf = unsafe { &*c };
-                Box::new(cf.start())
+                Box::new(Compat::new(cf.start()))
             }),
         };
 
         stream
             .then(move |message| {
                 match message {
                     Err(e) => Err(error!(message = "Error reading message from Kafka", error = ?e)),
-                    Ok(Err(e)) => Err(error!(message = "Kafka returned error", error = ?e)),
```

Comment:
Did I remote the right error message?

Reply:
If the error would still to be logged, I think it is fine to not have "Kafka returned error" in the beginning of it.

Reply:
The dilemma is between `Error reading message from Kafka` (from above) vs `Kafka returned error` (which I removed). I think the semantics have changed, and I should've removed `Error reading message from Kafka` and leave `Kafka returned error`.

Reply:
I agree that the semantics changed, so previously `Error reading message from Kafka` could have appeared if `MessageStream::poll()` returned an error and `Kafka returned error` could have appeared if the poll was successful, but returned [`KafkaError`](https://docs.rs/rdkafka/0.23.1/rdkafka/error/enum.KafkaError.html).

On the other hand, I'm in doubt now is `Kafka returned an error` the best description of any possible here error kind from the `KafkaError` enum. For example, it can be either [`MessageConsumption`](https://docs.rs/rdkafka/0.23.1/rdkafka/error/enum.KafkaError.html#variant.MessageConsumption) or [`Nul`](https://docs.rs/rdkafka/0.23.1/rdkafka/error/enum.KafkaError.html#variant.Nul), and in case of `Nul` it technically happens inside `librdkafka` library, but not returned by the Kafka cluster itself.

So I think actually the version `Error reading message from Kafka` is actually more general than `Kafka returned an error`, so it might make sense to keep it or maybe come up with some new message title which would be even more general.

Reply:
Thanks, I'll leave it as is then!